src/Processors/Transforms/Streaming/WatermarkStamper.h

```diff
@@ -66,63 +60,88 @@ SERDE class WatermarkStamper
 
     void processAfterUnmuted(Chunk & chunk);
 
-    bool requiresPeriodicOrTimeoutEmit() const { return periodic_interval || timeout_interval; }
+    bool requiresPeriodicOrTimeoutEmit() const { return params.periodic_interval || params.timeout_interval; }
 
     VersionType getVersion() const;
 
-    virtual void serialize(WriteBuffer & wb) const;
-    virtual void deserialize(ReadBuffer & rb);
+    void serialize(WriteBuffer & wb) const;
+    void deserialize(ReadBuffer & rb);
 
 protected:
-    virtual VersionType getVersionFromRevision(UInt64 revision) const;
+    VersionType getVersionFromRevision(UInt64 revision) const;
 
-private:
-    template <typename TimeColumnType, bool apply_watermark_per_row>
+    /// \brief Process events and apply watermark
     void processWatermark(Chunk & chunk);
 
-    void processPeriodic(Chunk & chunk);
-
-    void processTimeout(Chunk & chunk);
+    /// \brief Emit a watermark according to the EmitMode
+    /// \p old_watermark_ts - saved watermark before current processing
```

Comment:
Let's consistently use either `@brief, \brief, \param etc`, `\param` has builtin support by CLION

Reply:
do it in a separate PR